src/MadLibs.php

```diff
@@ -0,0 +1,169 @@
+        $finder->files()->in('src/data')->name($this->mad_libs_filename);
+
+        # if file not found, throws FileNotFoundException
+        if ($finder->hasResults()) {
+            // iterate through Finder object to retrieve the file
+            foreach ($finder as $file) {
+
+                if ($file->getFilename() == $this->mad_libs_filename) {
+                    $madLibsFileContents = $file->getContents();
+                    $madLibsArray = json_decode($madLibsFileContents, true);
+                    $madLibsOptions = array(); # to be: array of indexed mad libs titles
+
+                    // build madLibsOptions array of indexed titles
+                    $keyIndex = 0;
+                    foreach ($madLibsArray as $madLib) {
+                        $value = $madLib["title"];
+                        $option = array($keyIndex => $value);
+
+                        $madLibsOptions = array_merge($madLibsOptions, $option);
+                        $keyIndex++;
+                    }
+
+                    // print the titles and indices of the mad libs to the console
+                    $this->printTitleAndOptions($input, $output, $madLibsOptions);
+
+                    // PROMPT USER TO ENTER INDEX OF MAD LIBS SELECTION
+                    // CHECK IF THE SELECTED INDEX IS WITHIN THE RANGE
+                    $selectionIndex = -1;
+                    while ($selectionIndex < 0 || $selectionIndex > count($madLibsOptions)) {
+                        $selectionIndex = $this->getSelection($input, $output);
+                        if ($selectionIndex < 0 || $selectionIndex >= count($madLibsOptions)) {
+                            $output->writeln('<error>ERROR: You must enter a value between [0] and '
+                                . (count($madLibsOptions) - 1) . '. Please try again.</error>');
+                        }
+                    }
+
+                    $selection = $madLibsArray[$selectionIndex]; # the mad lib selected by the user
+                    $blanks = (array)$selection["blanks"]; # array of the selected mad lib blanks
+
+                    // PROMPT USER TO FILL IN EACH INDIVIDUAL BLANK
+                    $this->printSeparator($input, $output);
+                    $responses = $this->getResponses($input, $output, $blanks);
+
+                    // BUILD AND PRINT THE COMPLETED MAD LIB STRING
+                    // NOTE: The following assumes that a blank is always the second item in a mad lib.
+                    $values = $selection["value"]; # mad lib non-blanks, titled 'values' in madlibs.json
+                    $result = $this->buildMadLib($responses, $values);
+                    $this->printSeparator($input, $output);
+                    $output->writeln($result);
+
+                    if ($this->getPlayAgainResponse($input, $output) == '1') {
+                        $this->execute($input, $output);
+                    }
+                }
+            }
+        } else {
```

Comment:
To avoid the if/else you could do `if (!$finder->hasResults())` and then the main contents can be at the main level of the method. This is a strategy to avoid having to have nested code and keep conditionals at the top to be able to exit early.